src/main/java/GameLogic.java

```diff
@@ -29,33 +53,44 @@ public int convertInputToGridSquare(String input) {
         return inputConverted;
     }
 
-    public void makeMove(String input) {
+    private void makeMove(String input) {
         int inputConverted = convertInputToGridSquare(input);
         markSquare(inputConverted, activePlayer.getMark());
     }
 
-    public void markSquare(int squareNumber, Mark mark) {
-        grid.markSquare(squareNumber,mark);
+    private void markSquare(int squareNumber, Mark mark) {
+        grid.markSquare(squareNumber, mark);
     }
 
     public boolean gameIsWon() {
         return grid.winningLineExistsInGrid();
     }
 
-    public boolean moveIsNotLegal(String input) {
-        int inputConverted = convertInputToGridSquare(input);
-        return grid.moveIsNotLegal(inputConverted);
+    public boolean moveNotLegal(int input) {
+        return grid.moveNotLegal(input);
     }
 
     public boolean gameOngoing() {
-        if (grid.isFull() || gameIsWon()) return false;
-        else return true;
+        return !gameTied() && !gameIsWon();
+    }
+
+    private boolean gameTied() {
+        return grid.isFull();
     }
 
     public String generateComputerInput() {
+        int potentialInput = 0;
+        boolean moveIllegalTrue = true;
+        while (moveIllegalTrue) {
+            potentialInput = generateRandomComputerNumber();
+            moveIllegalTrue = moveNotLegal(potentialInput);
```

Comment:
Does this mean that the variable `moveIlliegalTrue` could be false? Confusing! I would just name it `illegalMove`.